psl-clojure/src/psl_clojure/core.clj

```diff
@@ -0,0 +1,401 @@
+  `(do
+     (.add ~model                     ; Add the predicate to the model
+           (java.util.HashMap.
+            (assoc ~argmap "predicate" (name '~pred-name)))) ; Add 'predicate'
+     (defn ~pred-name [& ~'args]    ; A function returning a QueryAtom
+       (QueryAtom.
+        (.getPredicate ~model (name '~pred-name)) ; Predicate from name
+        (into-array                               ; Array of Terms
+         Term
+         (for [~'a ~'args]
+           (if (symbol? ~'a) (Variable. (name ~'a)) ~'a)))))))
+
+(defmacro add-pred-constraint
+  "Add a predicate constraint to the model."
+  [model predicate constraint-type cname]
+  `(do
+     (.addConstraint
+      ~model
+      ~constraint-type
+      {"on" (.getPredicate ~model (name '~predicate))
+       "name" ~cname})))
+
+(defn add-rule [model formula weight squared name]
+  "Add a logical rule to the model."
+  (let [rule (WeightedLogicalRule. formula weight squared name)]
+    (.addRule model rule)))
+
+(defn add-rule-string
+  "Add a rule formatted as a string to the model."
+  ([model datastore rule-string weight squared]
+   (let [partial-rule (ModelLoader/loadRulePartial datastore rule-string)
+         rule (do (assert (not (.isRule partial-rule)))
+                  (.toRule partial-rule weight squared))]
+     (.addRule model rule)))
+  ([model datastore rule-string]
+   (let [partial-rule (ModelLoader/loadRulePartial datastore rule-string)
+         rule (do (assert (.isRule partial-rule))
+                  (.toRule partial-rule))]
+     (.addRule model rule))))
+
+;;; =============== Other functions for using PSL ====================
+
+(defn close-db
+  "Close Database, catching an IllegalStateException if it is already closed."
+  [database]
+  (try
+    (.close database)
+    (catch IllegalStateException e (str "caught exception: " (.getMessage e)))))
+
+(defn close-open-dbs "Close all open databases and returns their count."
+  [datastore]
+  (let [dbs (.getOpenDatabases datastore)
+        num (count dbs)]
+    (doseq [d dbs] (close-db d))
+    num))
+
+(defn default-inference
+  "Return an app for MPE inference using the configuration."
+  [model database config-bundle]
+  (LazyMPEInference. model database config-bundle))
+
+(defn ground-rules-by-name
+  "Return all ground rules from a given collection with the given name."
+  [ground-rules rule-name]
+  (for [gr ground-rules :when (= (.getName (.getRule gr)) rule-name)] gr))
+
+(defn ground-rules-by-name-sort
+  "Return a list of ground rules sorted by rule name."
+  [ground-rules]
+  (sort (fn [g1 g2]
+          (let [n1 (.getName (.getRule g1))
+                n2 (.getName (.getRule g2))]
+            (compare n1 n2)))
+        (seq ground-rules)))
+
+(defn ground-rules-names
+  "Return a list of distinct rule names associated with the given
+  collection of ground rules."
+  [ground-rules]
+  (sort (distinct (for [gr ground-rules] (.getName (.getRule gr))))))
+
+(defn ground-rules-print-summary
```

Comment:
This method is fine, but I want to make a greater comment on methods like these.

Typically, we want our interfaces to be shallow.
We want to do a minimal amount of effort to massage the data/interface into the correct format for the core PSL engine.
So methods like this are fine in experimental while we are figuring out what works and what doesn't.
But before this get upgraded to non-experimental, these methods will need to be in core and this interface provide a thin layer to them.